### PR TITLE
Move build to gh actions

### PR DESCRIPTION
**Pull request by danreeves**

### CODE REVIEW

**Comment by Coconutcoo** [comment]
Would it make more sense to just create a static/ or assets/ folder, and move the icons there, and then have them copied over during the build? Then you can just gitignore the whole extensions folder.

Thinking about when we look at adding weapon icons, it'd be nice to not have to put them into extensions directly, and conversely if we end up generating further files there, we don't need to gitignore each of them by themselves?

**Comment by danreeves** [comment]
The manifest and icon files need to be in the extension directory still, can't ignore them. And the manifest needs to reference the built files in order to load them/allow them to be loaded

**Comment by danreeves** [comment]
i think the number of generated files in the extension directory should be limited so i think it's fine gitignoring them individually for now rather than complicate the build process by copying files around

**Comment by danreeves** [comment]
PRs now include build artifacts if you want the latest extension build from one, e.g. https://github.com/danreeves/dt-exchange/actions/runs/3839130258

**Comment by Coconutcoo** [comment]
Oh gotcha - I thought the manifest was being generated too, for some reason I thought it was pulling info from elsewhere. Fully agree with you now :)

File: .github/workflows/pre-release.yml [hunk 1]
```diff
@@ -17,6 +17,14 @@ jobs:
       - name: "Checkout code"
         uses: "actions/checkout@v3.2.0"
 
+      - name: "Build"
+        uses: actions/setup-node@v3
+        with:
+          node-version: current
```
Comment (Coconutcoo) [inline .github/workflows/pre-release.yml]: Are we sure we want to be on current vs lts/*?

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    runs-on: "ubuntu-latest"
+
+    steps:
+      - name: "Checkout code"
+        uses: "actions/checkout@v3.2.0"
+
+      - name: "Build"
+        uses: actions/setup-node@v3
+        with:
+          node-version: current
```
Comment (Coconutcoo) [inline .github/workflows/pull-request.yml]: Same as above

**Review by Coconutcoo — COMMENTED** [review]
Other than pinning the node version, this looks good to me - local dev still works as expected

**Review by Coconutcoo — APPROVED** [review]

**Pull request merged**